micro-rdk/src/esp32/encoder.rs

```diff
@@ -94,9 +131,16 @@ where
         cfg: ConfigType,
         _: Vec<Dependency>,
     ) -> Result<EncoderType, EncoderError> {
-        let pin_a_num = cfg.get_attribute::<i32>("a")?;
+        let pins = cfg
+            .get_attribute::<Esp32EncoderConfig>("pins")
+            .map_err(EncoderError::EncoderConfigAttributeError)?;
 
-        let pin_b_num = cfg.get_attribute::<i32>("b")?;
+        let pin_a_num = pins.a.ok_or(EncoderError::EncoderConfigurationError(
+            "no 'a' pin specified".to_string(),
```

Comment:
before we would let the config parsing report the different errors (usually), why do we have a new error type?